mathics/builtin/assignment.py

```diff
@@ -65,376 +74,577 @@ def get_symbol_list(list, error_callback):
-                return False
-        elif lhs_name == "System`$MinPrecision":
-            # $MinPrecision = Infinity is not allowed
-            if rhs_int_value is not None and rhs_int_value >= 0:
-                ignore_protection = True
-                max_prec = evaluation.definitions.get_config_value("$MaxPrecision")
-                if max_prec is not None and max_prec < rhs_int_value:
-                    evaluation.message(
-                        "$MinPrecision", "preccon", Symbol("$MinPrecision")
-                    )
-                    return True
-            else:
-                evaluation.message(lhs_name, "precset", lhs, rhs)
-                return False
-        elif lhs_name == "System`$MaxPrecision":
-            if (
-                rhs.has_form("DirectedInfinity", 1)
-                and rhs.leaves[0].get_int_value() == 1
-            ):
-                ignore_protection = True
-            elif rhs_int_value is not None and rhs_int_value > 0:
-                ignore_protection = True
-                min_prec = evaluation.definitions.get_config_value("$MinPrecision")
-                if min_prec is not None and rhs_int_value < min_prec:
-                    evaluation.message(
-                        "$MaxPrecision", "preccon", Symbol("$MaxPrecision")
-                    )
-                    ignore_protection = True
-                    return True
-            else:
-                evaluation.message(lhs_name, "precset", lhs, rhs)
-                return False
 
-        # To Handle `OptionValue` in `Condition`
-        rulopc = Rule(
-            Expression(
-                "OptionValue",
-                Expression("Pattern", Symbol("$cond$"), Expression("Blank")),
-            ),
-            Expression("OptionValue", lhs.get_head(), Symbol("$cond$")),
+def process_assign_context_path(self, lhs, rhs, evaluation, tags, upset):
+    lhs_name = lhs.get_name()
+    currContext = evaluation.definitions.get_current_context()
+    context_path = [s.get_string_value() for s in rhs.get_leaves()]
+    context_path = [
+        s if (s is None or s[0] != "`") else currContext[:-1] + s for s in context_path
+    ]
+    if rhs.has_form("List", None) and all(valid_context_name(s) for s in context_path):
+        evaluation.definitions.set_context_path(context_path)
+        return True
+    else:
+        evaluation.message(lhs_name, "cxlist", rhs)
+        raise AssignmentException(lhs, None)
+
+
+def process_assign_minprecision(self, lhs, rhs, evaluation, tags, upset):
+    lhs_name = lhs.get_name()
+    rhs_int_value = rhs.get_int_value()
+    # $MinPrecision = Infinity is not allowed
+    if rhs_int_value is not None and rhs_int_value >= 0:
+        max_prec = evaluation.definitions.get_config_value("$MaxPrecision")
+        if max_prec is not None and max_prec < rhs_int_value:
+            evaluation.message("$MinPrecision", "preccon", Symbol("$MinPrecision"))
+            raise AssignmentException(lhs, None)
```

Comment:
I'm not sure, but I think that instead of `evaluation.message` and `AssignmentException` we could use `evaluation.error`.

Reply:
Maybe. I just try to keep the logic of the code as similar as possible to the original in this stage. We can change that later.

Reply:
Ok.